src/ui.jsx

```diff
@@ -0,0 +1,392 @@
+/**
+ * ui.jsx
+ *
+ * A file describing the UI of the application using React. A lot of the state accesses occur on the state property
+ * which apparently isn't the best practice but it gets rid of lots of unnecessary re-renders.
+ *
+ */
+"use strict";
+/*eslint-disable no-console, no-underscore-dangle*/
+
+import React from "react";
+
+import { executeInstruction, getFancyPositionInformation } from "./robot.js";
+import { parseInstructions, prepareRobots } from "./instructionreader.js";
+
+export default class MartianRobots extends React.Component {
+
+    render() {
+
+        return (
+            <div id="appContainer">
+                <Header />
+                <World />
+            </div>
+        );
+
+    }
+
+}
+
+/**
+ * The header bar at the top of the page.
+ */
+class Header extends React.Component {
+
+    render() {
+        return (
+            <div id="header">
+                <p>Martian Robots</p>
+            </div>
+        );
+    }
+
+}
+
+/**
+ * State entirely resides here; if it's need elsewhere, it can be passed down to child components as props.
+ */
+class World extends React.Component {
+
+    constructor() {
+
+        super();
+
+        /**
+         * State becomes immutable with some easily updatable properties. Every tick of the world updates the entire
+         * world state so that React can re-render it.
+         * @type {Object}
+         */
+        this.state = {
+
+            // Every robot that exists in the world and its state
+            robots: [],
+
+            // The state of the planet
+            planet: {
+                scents: {
+                    // E.g. "5,2": true
+                },
+                rows: 0,
+                cols: 0
+            },
+
+            outputMessages: []
+
+        };
+
+        this.setIntervalId = -1;
+
+        // TODO: Used for debugging, push state to this array after every tick so we can examine it later
+        this.savedStates = [];
```

Comment:
You probably don't need this. When debugging React state, you can literally do this:

```js
  render() {
    console.log(this.state);
    ...
  }
```

And it'll give you a great idea of what's going on, every time the state changes.